src/Microsoft.OData.Edm/Csdl/Serialization/EdmModelCsdlSchemaXmlWriter.cs

```diff
@@ -24,12 +25,14 @@ internal class EdmModelCsdlSchemaXmlWriter : EdmModelCsdlSchemaWriter
     {
         protected XmlWriter xmlWriter;
         protected readonly string edmxNamespace;
+        private readonly CsdlXmlWriterSettings writerSettings;
 
-        internal EdmModelCsdlSchemaXmlWriter(IEdmModel model, XmlWriter xmlWriter, Version edmVersion)
+        internal EdmModelCsdlSchemaXmlWriter(IEdmModel model, XmlWriter xmlWriter, Version edmVersion, CsdlXmlWriterSettings csdlXmlWriterSettings)

```

Comment:
Since this class is internal, maybe we should guarantee that the `CsdlXmlWriterSettings` parameter is not null.
Pass `new CsdlXmlWriterSettings()` where it's not externally supplied and check that the argument is not null in this constructor.

Reply:
I believe this is what's happening,

Reply:
@ElizabethOkerio There's a test where you're passing null as `CsdlXmlWriterSettings`. I suggest that you add a `Debug.Assert` statement to verify